lib/semiquo.gi

```diff
@@ -51,10 +51,19 @@ function(cong)
     SetOne(Q, One(S)^QuotientSemigroupHomomorphism(Q));
   fi;
 
-  if HasGeneratorsOfSemigroup(S) then
-    Qgens:= List( GeneratorsOfSemigroup( S ),
-     s -> s^QuotientSemigroupHomomorphism(Q));
-    SetGeneratorsOfSemigroup( Q, Qgens );
+  # The code below is safe, because GeneratorsOfSemigroup
+  # is synonymous with GeneratorsOfMagma, and there is a
+  # method for GeneratorsOfMagma if we know
+  # GeneratorsOfMagmaWithInverses.
+  # 
+  # We put this in, because groups do not set their 
+  # semigroup generators.
```

Comment:
It describes the reason *why* this hack is there in the first place: groups do not know their semigroup generating set, they only know their GeneratorsOfMagmaWithInverses.

I don't know whether putting Semigroup and Monoid generating sets into Group constructors wreaks more havoc on the library than I want to fix right now.

Reply:
Actually, after reading this added "explanation" I still do not quite understand what you mean. What is "this" in "We put this in" referring to: the check? Or the code inside the if/fi? Each interpretation seems valid to me (if one does not study resp. know the surrounding code well, at least). I first thought the latter, but now I think perhaps the former is meant?

After reading the surrounding code, and also the discussion in the PR, I think I now have a guess of what was meant, and wrote the following suggestion for an alternate comment based on that -- did I get it right?
> Ensure GeneratorsOfSemigroup(Q) is set if possible, by mapping GeneratorsOfSemigroup(S).
> Note that it is not sufficient to check HasGeneratorsOfSemigroup(S), because e. g. groups by default
> have GeneratorsOfMagmaWithInverses set but not GeneratorsOfSemigroup. However, there are
> GeneratorsOfSemigroup methods for domains knowing GeneratorsOfMagmaWithInverses or
> GeneratorsOfMagma, thus the following check and code works correctly even for groups.